### ISSUE TITLE
removeAllForClientId stops when finding any item with a null refresh token.

### ISSUE THREAD

**Issue by ill** [AzureAD/azure-activedirectory-library-for-objc#1162]
I am trying to cleanly log a user out and am calling removeAllForClientId.

However I see our token store has multiple entries for the same client id and only one has a refresh token.  I'm not sure what the others are for but it's preventing a clean deletion.  It's possible it's because our Staffhub app is part of the family of client ids as the rest of Microsoft Office.

The problem is this:

```
- (BOOL)removeItem:(nonnull ADTokenCacheItem *)item
             error:(ADAuthenticationError * __nullable __autoreleasing * __nullable)error
{
    RETURN_NO_ON_NIL_ARGUMENT(item);
    
    [item logMessage:@"Removing" level:ADAL_LOG_LEVEL_INFO correlationId:nil];

    OSStatus deleteStatus = [self deleteItem:item error:error];
    
    //if item does not exist in cache or does not contain a refresh token, deletion is enough and should return.
    if (deleteStatus != errSecSuccess 
        ///////////// problem at next line
        || [NSString adIsStringNilOrBlank:item.refreshToken])  //<---Problem is here, refresh token is nil so this causes the function to return NO
    {
        return [ADKeychainTokenCache checkStatus:deleteStatus operation:@"delete" correlationId:nil error:error];
    }

    return [self saveWipeTokenData:error];
}
```

My other option is to manually copy the function removeAllForClientId and iterate over all items to delete them.  I'm wondering if the removeAllForClientId should just continue deleting all items even if one removeItem call returns NO.

**Comment by oldalton** [comment]
Hi, @ill. Thanks for reporting the issue. It looks like a regression that we introduced in the previous release. We need to fix it, it's not by design.

**Comment by oldalton** [comment]
@jasoncoolmax I believe we fixed this in 2.6.2?

**Comment by jasoncoolmax** [comment]
Yep :)
